contracts/GitcoinResolver.sol

```diff
@@ -43,6 +45,30 @@ contract GitcoinResolver is
   // List of addresses allowed to write to this contract
   mapping(address => bool) public allowlist;
 
+  // Score data for querying
+  struct Score {
+    uint32 score; // compacted uint value 4 decimal places
+    uint64 issuanceDate; // For checking the age of the stamp, without loading the attestation
+    uint64 expirationDate; // This makes sense because we want to make sure the stamp is not expired, and also do not want to load the attestation
+    uint32 scorerId; // would we need this ???
+  }
+
+  /**
+   * @dev A struct representing the passport score for an ETH address.
+   */
+  //  TODO: use same type from IGitcoinPassportDecoder ?
+  struct ScoreAttestation {
+    uint256 score;
+    uint32 scorerID;
+    uint8 decimals;
+  }
+
+  // Mapping of addresses to scores
+  mapping(address => Score) public scores;
+
+  // Mapping of active passport score schemas - used when storing scores to state
+  mapping(bytes32 => bool) public scoreSchemas;
```

Comment:
I am just wondering if it makes sense to have multiple identical schemas "registered" as score schemas?
If we are thinking about creating a new score schema, then this will probably be treated completely different, since decoding it will be different. So I don't see a use-case where having multiple identical schemas valid at the same time would not cause confusion, hence we would want to avoid this.

Reply:
hmm is this comment still relevant? Are you suggesting storing it as a constant? My thinking was that by allowing multiple schemas we could update the schema without having to update the contract, e.g. adding additional metadata to the score attestation.